## Week 1 673380285-2 นายพิสิษฐ์ ทรัพย์อุดมโชติ

Lab 1.1: Environment Setup & First Script

In [4]:
print("Hello People Im here with my first python command na kub")

Hello People Im here with my first python command na kub


Lab 1.2: Working with Variables and Basic Data Types
- Objective: Students will define variables of different data types, perform simple operations, and display results.


2. Declare and print variables:


In [7]:
Age = 20
GPA = 3.01
FavColor = "Green"
Student = True

print("Age = ",Age ,type(Age))
print("GPA = ",GPA,type(GPA))
print("Favorite_Color = ",FavColor,type(FavColor))
print("Student = ",Student,type(Student))

Age =  20 <class 'int'>
GPA =  3.01 <class 'float'>
Favorite_Color =  Green <class 'str'>
Student =  True <class 'bool'>


3. Perform simple arithmetic:


In [8]:
num1 = 10
num2 = 5

sum = num1 + num2
sub = num1 - num2
multi = num1 * num2
Dev = num1 / num2

print("Sum = ",sum)
print("Subtraction = ",sub)
print("multiplication = ",multi)
print("Devide = ",Dev)

Sum =  15
Subtraction =  5
multiplication =  50
Devide =  2.0


4. String concatenation and f-strings:


In [9]:
First_name = "Phisit"
Last_name = "Sapaudomchot"

Full_name = " My name is " + First_name + " " + Last_name

print(f"Hello,{Full_name}")

Hello, My name is Phisit Sapaudomchot


5. User input and type conversion:


In [2]:
num = input("What your favorite number : ")
print("type of your Number is : ", type(num))
num = int(num)
result = num + 5
print(num," + 5 = ", result)

type of your Number is :  <class 'str'>
20  + 5 =  25


Activity 1.1: Class Discussion - Uses of Scripting
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.


คำตอบ: ใช้ทำเป็นโปรแกรมเครื่องคิดเลขหรือการคำนวณสิ่งต่างๆตามต้องการ โดยปรับให้เข้ากับตัวเองได้เลย

ใช้ทำโปรเจค data analyze ได้

เขีบนเกมอยางง่ายได้ (XO งู)




Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [ ]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))


13.5
<class 'float'>


ได้ผลลัพธ์ : 13.5

ประเภทDataType : float

ทำไมถึงเป็นประเภทนี้ : ใน python ถ้าเอา int + float จะได้ผลลัพธ์ออกมาเป็น float ทันที


In [ ]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์ : Alice is 30 years old.

ประเภทDataType : String

ทำไมถึงเป็นประเภทนี้ : เป็น string เพราะตั้งต้นเป็น string เนื่องจากตอนประกาศ ใส่ ""

In [ ]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


ได้ผลลัพธ์ :
False
boolean

ประเภทDataType : boolean

ทำไมถึงเป็นประเภทนี้ : ค่า True/False เป็นประเภท Boolean

In [ ]:
# Puzzler 4
# What happens here?
# num_str = input("Enter a number: ")
# result = num_str + 5
# print(result)

ได้ผลลัพธ์: ไม่มีอะไรเกิดขึ้น เพราะโดนคอมเม้นไปหมดแล้ว เหมือนไม่มีอะไร

# คำนวณพื้นที่รูปเลขาคณิตพื้นฐานอย่างง่าย

In [3]:
print("Area of shapes")

print("1.Regtangle")
print("2.Circle")
print("3.Triangle")

Check = int(input("Enter a number: "))

print(" ")
if Check == 1:
    print("Find the Area of Regtangle")
    Wid = float(input("Enter a width : "))
    Hid = float(input("Enter a height : "))
    A = Wid * Hid
    print("Area is", A)
elif Check == 2:
    print("Find the Area of Circle")
    R = float(input("Enter a Radius : "))
    A = 3.14 * R * R
    print("Area is", A)
elif Check == 3:
    print("Find the Area of Triangle")
    B = float(input("Enter a Base : "))
    H = float(input("Enter a height : "))
    A = 0.5 * B * H
    print("Area is", A)

else:
    print("Invalid choice")




Area of shapes
1.Regtangle
2.Circle
3.Triangle
 
Find the Area of Regtangle
Area is 100.0
